### DISCUSSION TITLE
Writing a Custom Storage for YAML support

### DISCUSSION THREAD

**Discussion by TheNewThinkTank** [msiemens/tinydb#532] [Extensions]
Following official guidelines, https://tinydb.readthedocs.io/en/latest/extend.html , I attempt to add YAML support to my TinyDB project as follows (my complete script):

```Python
import yaml  # type: ignore

from tinydb import TinyDB  # type: ignore


class YAMLStorage(Storage):
    def __init__(self, filename):
        self.filename = filename

    def read(self):
        with open(self.filename) as handle:
            try:
                data = yaml.safe_load(handle.read())
                return data
            except yaml.YAMLError:
                return None

    def write(self, data):
        with open(self.filename, 'w+') as handle:
            yaml.dump(data, handle)

    def close(self):
        pass


if __name__ == "__main__":
    db = TinyDB('db.yml', storage=YAMLStorage)
```
And receive following name error (unknown base class):

```BASH
    class YAMLStorage(Storage):
                      ^^^^^^^
NameError: name 'Storage' is not defined
```
Is there some additional import I am missing perhaps?

**Comment by TheNewThinkTank** [comment]
For reference, here is the specific commit, with GitHub Actions workflow failing with the above mentioned error: https://github.com/TheNewThinkTank/fitness-tracker/commit/8bf84d3458cc0ffaeb35af6d725b7fe9e4064862

**Comment by TheNewThinkTank** [comment]
Solved by importing Storage from tinydb.storages

**Reply by msiemens** [reply]
Glad to hear to were able to solve this 🙂

**Reply by TheNewThinkTank** [reply]
Thanks, @msiemens , works like a charm now, and I swapped my project to using YAML-based TinyDB only
